environment/src/main/java/com/sequenceiq/environment/environment/service/EnvironmentCreationService.java

```diff
@@ -123,6 +126,21 @@ private Environment initializeEnvironment(EnvironmentCreationDto creationDto) {
         return environment;
     }
 
+    private void initializeEnvironmentTunnel(Environment environment) {
+        Tunnel tunnel = environment.getExperimentalFeaturesJson().getTunnel();
+        boolean ccmV2Enabled = entitlementService.ccmV2Enabled(environment.getAccountId());
+        if (Tunnel.CCMV2 == tunnel && !ccmV2Enabled) {
+            throw new BadRequestException("CCMV2 not enabled for account.");
+        } else if (Tunnel.CCM == tunnel && ccmV2Enabled) {
+            ExperimentalFeatures experimentalFeaturesJson = environment.getExperimentalFeaturesJson();
+            experimentalFeaturesJson.setTunnel(Tunnel.CCMV2);
+            environment.setExperimentalFeaturesJson(experimentalFeaturesJson);
+            LOGGER.info("Environment is initialized with CCMV2 tunnel.");
+        } else {
+            LOGGER.info("Environment is initialized with Direct tunnel.");
```

Comment:
that's not true, it could be cluster proxied. although customer won't start that kind of cluster

Reply:
you could fix this in a followup pr

Reply:
ccm is different from cluster proxy right ? I mean all requests go through clusterproxy but which tunnel used would be diffferent.

Reply:
talked about in private, but to have it here also. CB can connect with:
- DIRECT (no cluster proxy)
- cluster proxy, but without initializing tunnel via CCM, which means cluster proxy is connecting directly
- CCM, which means CB will initialize, configure cluster proxy with tunnel, and then use cluster proxy for communication

Reply:
Updated log statement.